tutorials/misc/plot_ecog.py

```diff
@@ -11,192 +11,343 @@
+#
 # Next, we'll compute the signal power in the gamma (30-90 Hz) and alpha
-# (8-12 Hz) bands.
-gamma_power_t = raw.copy().filter(30, 90).apply_hilbert(
-    envelope=True).get_data()
-alpha_power_t = raw.copy().filter(8, 12).apply_hilbert(
-    envelope=True).get_data()
-gamma_power = gamma_power_t.mean(axis=-1)
-alpha_power = alpha_power_t.mean(axis=-1)
+# (8-12 Hz) bands, downsampling the result to 10 Hz (to save time).
 
-###############################################################################
-# Now let's use matplotlib to overplot frequency band power onto the electrodes
-# which can be plotted on top of the brain from
-# :func:`~mne.viz.snapshot_brain_montage`.
+freqs = np.linspace(5, 95, 10)
+power = mne.time_frequency.tfr_multitaper(epochs, freqs, n_cycles=freqs / 2,
+                                          return_itc=False)
 
-# Convert from a dictionary to array to plot
-xy_pts = np.vstack([xy[ch] for ch in raw.info['ch_names']])
+###############################################################################
+# We can project gamma power from the sensor data to the nearest locations on
+# the pial surface and visualize that:
 
-# colormap to view spectral power
-cmap = 'viridis'
+# sphinx_gallery_thumbnail_number = 5
+
+sfreq = 10
+gamma_power_t = np.median(power.data[:, power.freqs >= 30], axis=1)
+evoked = mne.EvokedArray(gamma_power_t, epochs.info)
+
+# downsample for animation
+evoked.resample(sfreq)
+src = mne.read_source_spaces(
+    op.join(subjects_dir, 'fsaverage', 'bem', 'fsaverage-ico-5-src.fif'))
+trans = None  # identity transform
+stc = mne.stc_near_sensors(evoked, trans, 'fsaverage', src=src,
+                           mode='nearest', subjects_dir=subjects_dir,
+                           distance=0.02)
+vmin, vmax = np.percentile(gamma_power_t, [10, 90])
+clim = dict(kind='value', lims=[vmin * 0.9, vmin, vmax])
+brain = stc.plot(surface='pial', hemi='rh', colormap='viridis',
+                 clim=clim, views=['lat', 'med'], subjects_dir=subjects_dir,
+                 size=(600, 800), smoothing_steps=5)
 
-# Create a 1x2 figure showing the average power in gamma and alpha bands.
-fig, axs = plt.subplots(1, 2, figsize=(20, 10))
-# choose a colormap range wide enough for both frequency bands
-_gamma_alpha_power = np.concatenate((gamma_power, alpha_power)).flatten()
-vmin, vmax = np.percentile(_gamma_alpha_power, [10, 90])
-for ax, band_power, band in zip(axs,
-                                [gamma_power, alpha_power],
-                                ['Gamma', 'Alpha']):
-    ax.imshow(im)
-    ax.set_axis_off()
-    sc = ax.scatter(*xy_pts.T, c=band_power, s=200,
-                    cmap=cmap, vmin=vmin, vmax=vmax)
-    ax.set_title(f'{band} band power', size='x-large')
-fig.colorbar(sc, ax=axs)
+# You can save a movie like the one on our documentation website with:
+# brain.save_movie(time_dilation=1, interpolation='linear', framerate=5,
+#                  time_viewer=True)
```

Comment:
Points 1/2/4/5 are really bugs with Brain so I don't think we can/should fix them here. For the blockiness it's a choice to show the discretization of `src` but we could change the steps to `5` for simplicity. For 6/7 @alexrockhill you could update the narrative doc to mention that it's the one with highest-amplitude activation in the gamma band, and maybe change the view shown (but it's always kind of hard to see so we might just need to live with 7 as is)

Reply:
(although maybe for `2` changing the upper crop time limit or increasing `framerate` a tiny bit might help...)

Reply:
- Ah right, the blockiness is because this is `stc_near_sensors` not an MNE/dSPM/etc map. That makes sense.
- is it really not possible to pick the sphere+trace color?  If not, can we change the colormap to something (magma/inferno perhaps?) that will contrast with the default MPL blue of the sphere?

Reply:
I'd rather change the colormap to `inferno` or something similar.

I don't think there is a way with Brain currently to set which points get a trace, since currently it's meant to be done interactively (and have some sensible defaults). For a programmatic case you could currently do `brain.add_foci` then extract the time course manually and plot it but this is overkill for an example. We could probably make public the functions that internally are used to add a dot + trace but not in this PR.

Reply:
I'm fine with changing colormap (and maybe which brain view(s) are shown) to make the sphere easier to see.